client/src/pages/school/CreateReactTableForSchool.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import { useTable, useSortBy } from "react-table";
+import { Table, Tbody, Th, Thead, Tr, Td, chakra } from "@chakra-ui/react";
+import processSchoolRow from "./processSchoolRow";
+import { TriangleDownIcon, TriangleUpIcon } from "@chakra-ui/icons";
+
+const CreateReactTableForSchool = ({ schoolData }) => {
+	// Prepare the data for React Table
+	const reactTableColumns = React.useMemo(
+		() => [
+			{
+				Header: "Country",
+				accessor: "country", // accessor is the "key" in the data
+			},
+			{
+				Header: "Name",
+				accessor: "name",
+			},
+			{
+				Header: "Location",
+				accessor: "location",
+			},
+			{
+				Header: "Description",
+				accessor: "description",
+			},
+			{
+				Header: "Person Responsible",
+				accessor: "full_name",
+			},
+			{
+				Header: "Status",
+				accessor: "status",
+			},
+			{
+				Header: "Deployment Date",
+				accessor: "deploymentdate",
+				sortType: (a, z) => {
+					return (
+						a.values.deploymentdate_sortvalue -
+						z.values.deploymentdate_sortvalue
+					);
+				},
+			},
+			{
+				Header: "hidedate", // for sorting purposes only - see above
+				accessor: "deploymentdate_sortvalue",
+				isVisible: true,
+			},
+			{
+				Header: "Created At",
+				accessor: "created_at",
+			},
+		],
+		[]
+	);
+
+	const reactTableData = React.useMemo(
+		() => schoolData.map((element) => processSchoolRow(element)),
+		[schoolData]
+	);
+
+	const {
+		getTableProps,
+		getTableBodyProps,
+		headerGroups,
+		rows,
+		prepareRow,
+		setHiddenColumns,
+	} = useTable(
+		{
+			columns: reactTableColumns,
+			data: reactTableData,
+			hiddenColumns: reactTableColumns
+				.filter((column) => !column.isVisible)
+				.map((column) => column.accessor),
+		},
+		useSortBy
+	);
+
+	// According to the confusing React-Table documentation, this is the only way to hide a column
+	// See @ggascoigne AT https://github.com/TanStack/table/issues/1804
+	// https://codesandbox.io/s/relaxed-night-y5gcl?file=/src/App.js:980-985
+
+	React.useEffect(() => {
```

Comment:
Same as above but with useEffect instead of useMemo

Reply:
Done!